<!DOCTYPE html>
<html lang="en">
	<head>
		<meta charset="utf-8">
		<meta http-equiv="X-UA-Compatible" content="IE=edge,chrome=1">
		<meta name="viewport" content="width=device-width, initial-scale=1">
		 
			
  
    <meta name="twitter:card" content="summary"/>
    
      <meta name="twitter:image" content="/images/avatar.png" />
    
  
  
  <meta name="twitter:title" content="Molecular Evolution Tutorial"/>
  <meta name="twitter:description" content="First, of perhaps a few more tutorials. "/>
  
    <meta name="twitter:site" content="@wandering_jeff"/>
  
  
  
  
    <meta name="twitter:creator" content="@tijeco"/>
  



		
		<meta name="author" content="tijeco">
		<meta name="description" content="bioinformatics blog">
		<meta name="generator" content="Hugo 0.33" />
		<title>Molecular Evolution Tutorial &middot; tijeco</title>
		<link rel="shortcut icon" href="/images/favicon.ico">
		<link rel="stylesheet" href="/css/style.css">
		<link rel="stylesheet" href="/css/highlight.css">

		
		<link rel="stylesheet" href="/css/font-awesome.min.css">
		

		
		<link href="/index.xml" rel="alternate" type="application/rss+xml" title="tijeco" />
		

		
	</head>

    <body>
       <nav class="main-nav">
	
	
		<a href='/'> <span class="arrow">←</span>Home</a>
	
	<a href='/posts'>Archive</a>
	<a href='/tags'>Tags</a>
	

	

	
	<a class="cta" href="/index.xml">Subscribe</a>
	
</nav>


        <section id="wrapper" class="post">
            <article>
                <header>
                    <h1>
                        Molecular Evolution Tutorial
                    </h1>
                    <h2 class="headline">
                    Jan 20, 2018 00:00
                    · 345 words
                    · 2 minute read
                      <span class="tags">
                      
                      
                          
                              <a href="/tags/tutorial">tutorial</a>
                          
                              <a href="/tags/bioinformatics">bioinformatics</a>
                          
                              <a href="/tags/molecular-evolution">molecular evolution</a>
                          
                      
                      
                      </span>
                    </h2>
                </header>
                
                  
                    <div id="toc">
                      <nav id="TableOfContents">
<ul>
<li><a href="#hyphy">HYPHY</a></li>
<li><a href="#pervasive-positive-selection-in-cone-snail-venom">Pervasive positive selection in cone snail venom</a></li>
<li><a href="#paml">PAML</a>
<ul>
<li><a href="#serine-threonine-protein-kinase-gene-family">Serine/threonine-protein kinase gene family</a>
<ul>
<li><a href="#codeml-commands">CODEML commands</a></li>
<li><a href="#analyze-results">Analyze results</a></li>
</ul></li>
</ul></li>
</ul>
</nav>
                    </div>
                  
                
                <section id="post-body">
                    <p>This is a brief rundown of how to conduct tests for positive selection in sequence data using PAML and HYPHY</p>

<p></p>

<h1 id="hyphy">HYPHY</h1>

<ul>
<li>all files need for this tutorial are <strong><a href="https://github.com/tijeco/MolecularEvolutionTutorial/raw/master/CodeML.files.zip">here</a></strong></li>
</ul>

<h1 id="pervasive-positive-selection-in-cone-snail-venom">Pervasive positive selection in cone snail venom</h1>

<p><strong>1.</strong> Get the file <strong>Conus.01.fas</strong></p>

<p><em>This is our translated alignment</em></p>

<p><img src="http://www.tijeco.info/wp-content/uploads/2017/04/Superfamily-O1-1.png" alt="alt tag" /></p>

<p><strong>2.</strong> We will now do the FUBAR analysis
* This is a site specific method for detecting selection at the amino acid level, helpful in arms race scenarios</p>

<p><strong>3.</strong> Go to <a href="http://www.datamonkey.org/dataupload.php">http://www.datamonkey.org/dataupload.php</a></p>

<p><strong>4.</strong> Select  choose file, select <strong><em>Conus.01.fas</em></strong></p>

<ul>
<li>Then, press <strong><em>Upload</em></strong></li>
</ul>

<p><strong><em>5.</em></strong> If your data is no bueno, this is where it will let you know</p>

<ul>
<li><p>It accepts fasta alignments and nexus files</p></li>

<li><p>Also listed here should be information about the alignments</p></li>

<li><p>If all is well, press <strong><em>Proceed to the analysis menu</em></strong></p></li>
</ul>

<p><strong>6.</strong> Under <strong><em>Method</em></strong> select <strong><em>FUBAR</em></strong></p>

<p><strong>7.</strong> For now leave everything else alone and just press <strong><em>Run</em></strong></p>

<ul>
<li><p>If you had a nexus file, you could specify your own tree in Newick format</p></li>

<li><p>For now we&rsquo;ll just let it use a generated neighbor joining tree</p></li>
</ul>

<p><strong>8.</strong> The results will look like the following <img src="https://raw.githubusercontent.com/tijeco/MolecularEvolutionTutorial/master/fubarResults.png" alt="This file is gone now" /></p>

<ul>
<li>This is a heat map showing sites are under positive selection and negative selection</li>
</ul>

<hr />

<h1 id="paml">PAML</h1>

<h2 id="serine-threonine-protein-kinase-gene-family">Serine/threonine-protein kinase gene family</h2>

<ul>
<li>This has been adapted from a tutorial by <a href="evosite3d.blogspot.com">Romain Studer</a></li>
<li>In the evolution vertebrates, we would like to know if the branch leading to the Teleost fishes (genes A50 to A54)</li>
</ul>

<p>You will need the following files</p>

<p><strong>1. TF105351.Eut.3.phy</strong><br></p>

<ul>
<li>this is the alignment file</li>
</ul>

<p><img src="https://3.bp.blogspot.com/-LxFevYQBQ7I/TnN0vARVO4I/AAAAAAAAABM/PAxYkWUTQcw/s320/TF105351.Eut.3.aln.png" alt="alt tag" /></p>

<p><strong>2. TF105351.Eut.3.53876.tree</strong></p>

<ul>
<li>this is the newick tree with the branch of interest selected</li>
</ul>

<p><img src="https://2.bp.blogspot.com/-QCyFGC6o2zQ/Uad84nz-S3I/AAAAAAAAAEg/W12U2V0CizQ/s1600/Tree.png" alt="alt tag" /></p>

<p><strong>3. TF105351.Eut.3.53876.ctl</strong></p>

<ul>
<li>CodeML configuration file for <strong><em>alternative model</em></strong></li>
</ul>

<p><strong>4. TF105351.Eut.3.53876.fixed.ctl</strong></p>

<ul>
<li>CodeML configuration file for <strong><em>null model</em></strong></li>
</ul>

<h3 id="codeml-commands">CODEML commands</h3>

<pre><code class="language-bash">codeml TF105351.Eut.3.53876.ctl
codeml TF105351.Eut.3.53876.fixed.ctl
</code></pre>

<h3 id="analyze-results">Analyze results</h3>

<p>Get liklihood values</p>

<pre><code class="language-bash">grep lnL TF105351.Eut.3.53876.mlc
</code></pre>

<pre><code>lnL(ntime: 41  np: 46):  -4707.209701      +0.000000
</code></pre>

<p>Liklihood value for alternative model is <strong>-4707.209701</strong></p>

<pre><code>grep lnL TF105351.Eut.3.53876.fixed.mlc
</code></pre>

<pre><code>lnL(ntime: 41  np: 45):  -4710.222252      +0.000000
</code></pre>

<p>Liklihood value for null model is <strong>-4710.222252</strong></p>

<p>ΔLRT = 2×(lnL1 - lnL0) = 2×(-4707.209701 - (-4710.222252)) = 6.02578</p>

<p>The degree of freedom is 1 (np1 - np0 = 46 - 45).
p-value = 0.01098 (under χ²) =&gt; significant.</p>
                </section>
            </article>

            
                <a class="twitter" href="https://twitter.com/intent/tweet?text=%2fposts%2fmolecularevolutiontutorial%2f - Molecular%20Evolution%20Tutorial by @wandering_jeff"><span class="icon-twitter"> tweet</span></a>

<a class="facebook" href="#" onclick="
    window.open(
      'https://www.facebook.com/sharer/sharer.php?u='+encodeURIComponent(location.href),
      'facebook-share-dialog',
      'width=626,height=436');
    return false;"><span class="icon-facebook-rect"> Share</span>
</a>

            

            
                <div id="disqus_thread"></div>
<script type="text/javascript">
    var disqus_shortname = 'blog-tijeco-info'; 

     
    (function() {
        var dsq = document.createElement('script'); dsq.type = 'text/javascript'; dsq.async = true;
        dsq.src = '//' + disqus_shortname + '.disqus.com/embed.js';
        (document.getElementsByTagName('head')[0] || document.getElementsByTagName('body')[0]).appendChild(dsq);
    })();
</script>
<noscript>Please enable JavaScript to view the <a href="https://disqus.com/?ref_noscript">comments powered by Disqus.</a></noscript>
</div>

            

            

            
        </section>
        <footer id="footer">
    
        <div id="social">

	
	
    <a class="symbol" href="https://www.facebook.com/nodejh">
        <i class="fa fa-envelope"></i>
    </a>
    
    <a class="symbol" href="https://www.github.com/tijeco">
        <i class="fa fa-github"></i>
    </a>
    
    <a class="symbol" href="https://www.twitter.com/wandering_spider">
        <i class="fa fa-linkedin"></i>
    </a>
    
    <a class="symbol" href="https://www.twitter.com/wandering_spider">
        <i class="fa fa-twitter"></i>
    </a>
    


</div>

    
    <p class="small">
    
       © Copyright 2018 <i class="fa fa-heart" aria-hidden="true"></i> tijeco
    
    </p>
    
</footer>


        <script src="/js/jquery-2.2.4.min.js"></script>
<script src="/js/main.js"></script>
<script src="/js/highlight.min.js"></script>
<script>hljs.initHighlightingOnLoad();</script>




  
<script>
(function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
(i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
})(window,document,'script','https://www.google-analytics.com/analytics.js','ga');

ga('create', 'your_google_analytics_id', 'auto');
ga('send', 'pageview');
</script>





    </body>
</html>
